source/MaterialXRuntime/RtSchema.h

```diff
@@ -126,6 +126,11 @@ class RtSchemaBase
         return getPrim().getMetadata(name, type);
     }
 
+    virtual RtTokenVec getEditableMetadata() const
```

Comment:
As mentioned above return a const ref. An empty list can be keep as a local object in RtSchema.cpp.

Reply:
We need a better name for the method, as discussed this is not about editability. One possible name is getPublicMetadata(), as opposed to internal/custom metadata being considered "private". Not sure, what do you think?

Reply:
We need this on ports as well no? For example colorspace and units can be set on ports.

This is more complicated, since ports are not schemas, but we still need this to be schema specific. So I think we need a separate method on RtSchema for querying this on its ports: getPortPublicMetadata(const RtToken& port).

Reply:
I'm a touch confused about the getPortPublicMetadata method. In the getPublicMetadata case, the metadata names returned are a fixed list for all nodegraphs/nodes. So shouldn't the list also be fixed for all ports? So shouldn't it be just getPortPublicMetadata() with no passed in port name? Also in this case, what is this data? Is it xpos/ypos/width/height/uicolor/uivisible/uiadvanced? At the moment, I don't really support changing all those values, but I suppose they would make sense as changeable values.

Reply:
colorspace and unit might be other such inputs

Reply:
Yes it's all metadata that has been defined by the spec for the various inputs. And the problem is that it's not the same for all inputs. For example colorspace is only defined on inputs of type color or filename.

Reply:
So this gets more complicated.. But if the API should support listing all spec-defined metadata we need to define this on both prims/schemas and ports. As far as I understand this is what Bernard wanted.